### DISCUSSION TITLE
Plugins Developement

### DISCUSSION THREAD

**Discussion by AVariot** [netbox-community/netbox#13356] [Help Wanted!] [answered]
Hi ! I'm working on netbox plugin but i have errors with the view edit, but i don't know why ?
I got this : 
![image](https://github.com/netbox-community/netbox/assets/91867982/204e6593-42d1-4d8f-9844-98b575cea41e)
 
this is my views.py :
```
class FournisseursEditView(generic.ObjectEditView):
    queryset = models.FNSModel.objects.all()
    form = forms.FournisseursForm
```
and my models.py :
```
class FNSModel(NetBoxModel):
    name = models.CharField(
        max_length=50
    )
    url = models.URLField(
        blank=True
    )

    def __str__(self):
        return (self.name)

    def get_absolute_url(self):
        return reverse('plugins:netbox_fournisseurs_commandes:fnsmodel', args=[self.pk])
```

Thank you for your help

**Comment by abhi1693** [comment, marked as answer]
You need to also write code for API integration. You are missing the serializer. I suggest you refer to the plugin tutorial for more understanding https://github.com/netbox-community/netbox-plugin-tutorial

**Reply by AVariot** [reply]
Thank you for your response, i will try it!

**Reply by AVariot** [reply]
I have the same error

**Reply by candlerb** [reply]
You can look in your browser's developer console to find out what API calls are being made. It could be that the form is making AJAX calls in the background, e.g. to populate a picklist.

Otherwise, show the full backtrace, which shows what series of function calls through Netbox is ending up at netbox/utilities/api.py

There could be other things wrong with your plugin, e.g. your [url registration](https://docs.netbox.dev/en/stable/plugins/development/views/#url-registration).

You could also start with the tutorial plugin, check that works, and then model your plugin on that.

**Reply by AVariot** [reply]
Actually it's working but idk why, sorry
